packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -923,6 +1069,43 @@ export function pushSegmentFinale(
   }
 }
 
+function pushViewTransitionAttributes(
+  target: Array<Chunk | PrecomputedChunk>,
+  formatContext: FormatContext,
+): void {
+  if (!enableViewTransition) {
+    return;
+  }
+  const viewTransition = formatContext.viewTransition;
+  if (viewTransition === null) {
+    return;
+  }
+  if (viewTransition.name !== 'auto') {
+    pushStringAttribute(
+      target,
+      'vt-name',
+      viewTransition.nameIdx === 0
+        ? viewTransition.name
+        : viewTransition.name + '_' + viewTransition.nameIdx,
+    );
+    // Increment the index in case we have multiple children to the same ViewTransition.
+    // Because this is a side-effect in render, we should ideally call pushViewTransitionAttributes
+    // after we've suspended (like forms do), so that we don't increment each attempt.
+    // TODO: Make this deterministic.
+    viewTransition.nameIdx++;
+  }
+  pushStringAttribute(target, 'vt-update', viewTransition.update);
+  if (viewTransition.enter !== null) {
+    pushStringAttribute(target, 'vt-enter', viewTransition.enter);
+  }
+  if (viewTransition.exit !== null) {
+    pushStringAttribute(target, 'vt-exit', viewTransition.exit);
+  }
+  if (viewTransition.share !== null) {
+    pushStringAttribute(target, 'vt-share', viewTransition.share);
+  }
```

Comment:
We've tried hard to keep the output spec compliant where possible. squatting on a data namespace isn't as culturally problematic as squatting on a potential spec namespace. I think we should eat the bytes and use `data-vt-*`. It's not likely meaningfully incompatible with user code

Reply:
People enumerate over the `data-` set a lot. We lived with it for `<link>` and `<style>` for precedence since who would add `data-` to those but you enumerate `dataSet` a lot on your own owned elements. So I think `data-` is a no go. Can't add stuff to that.

Reply:
Maybe `r-vt-` as the namespace? `r-` also seems unlikely to be used in the future

Reply:
`x-` will never be added to HTML so `x-vt-*` might be the closest thing to compliant. It’s still technically not but at least we know it’ll never block future attributes